src/components/RepoInputer.vue

```diff
@@ -0,0 +1,188 @@
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, reactive, ref, watch } from "vue";
+import { useStore } from "vuex";
+import { GITHUB_REPO_URL_REG } from "../helpers/consts";
+import toast from "../helpers/toast";
+
+interface State {
+  repo: string;
+}
+
+export default defineComponent({
+  name: "RepoInputer",
+  setup() {
+    const store = useStore<AppState>();
+    const state = reactive<State>({
+      repo: "",
+    });
+    const inputElRef = ref<HTMLInputElement | null>(null);
+
+    const handleAddRepoBtnClick = () => {
+      if (store.state.isFetching) {
+        return;
+      }
+
+      let rawRepos = state.repo;
+      if (rawRepos === "" && store.state.repos.length === 0) {
+        rawRepos = "bytebase/star-history";
+      }
+
+      if (rawRepos === "") {
+        toast.warn("Please input the repo name");
+        return;
+      }
+
+      for (const rawRepo of rawRepos.split(" ")) {
+        let repo = rawRepo;
+        if (GITHUB_REPO_URL_REG.test(repo)) {
+          const regResult = GITHUB_REPO_URL_REG.exec(repo);
+          if (regResult && regResult[1]) {
+            repo = regResult[1];
+          }
+        }
+        if (store.state.repos.includes(repo)) {
+          toast.warn(`Repo ${repo} is already on the chart`);
+          continue;
+        }
+        store.commit("addRepo", repo);
+      }
+      state.repo = "";
+    };
+
+    const handleRepoItemClick = (repo: string) => {
+      store.commit("delRepo", repo);
+    };
+
+    const handleInputerKeyDown = (event: KeyboardEvent) => {
+      if (event.key === "Enter") {
+        event.preventDefault();
+        handleAddRepoBtnClick();
+      }
+    };
+
+    const handleInputerPasted = async (event: ClipboardEvent) => {
+      if (!inputElRef.value) {
+        return;
+      }
+      const inputEl = inputElRef.value;
+      if (event.clipboardData) {
+        event.preventDefault();
+        const text = event.clipboardData
+          .getData("text")
+          .replace(/(\r\n|\n|\r)/gm, "");
+        const value = state.repo;
+        const prevStr = value.slice(
+          0,
+          Math.min(inputEl.selectionStart || 0, inputEl.selectionEnd || 0)
+        );
+        const nextStr = value.slice(
+          Math.max(inputEl.selectionStart || 0, inputEl.selectionEnd || 0)
+        );
+        state.repo = `${prevStr}${text}${nextStr}`;
+      }
+    };
+
+    const handleClearAllRepoBtnClick = () => {
+      store.commit("clearAll");
+    };
+
+    watch(
+      () => store.state.repos,
+      () => {
+        let hash = "";
+        if (store.state.repos.length > 0) {
+          hash = `#${store.state.repos.join("&")}`;
+        }
+        window.location.hash = hash;
+      }
+    );
+
+    return {
+      state,
+      inputElRef,
+      isFetching: computed(() => {
+        return store.state.isFetching;
+      }),
+      repos: computed(() => {
+        return store.state.repos;
+      }),
```

Comment:
move the `computed` part below the `ref`, it's some state of something you want use.

keep `return` clean. maybe computed logic is too huge.